todo/src/todocli.rs

```diff
@@ -137,23 +123,24 @@ mod tests {
     use factori::create;
 
     use super::{mocks::*, *};
-    use crate::{repository::file_storage::MockStorage, terminal::MockUserInterface};
+    use crate::{repository::todo::MockStorage, terminal::MockUserInterface};
 
     #[tokio::test]
     async fn should_add_todo() {
         let mut mock_storage = MockStorage::new();
         let mut mock_user_interface = MockUserInterface::new();
-        let todo_added = Todo::new("todo 3".to_string());
+        let id = Uuid::new_v4();
+        let todo_added = Todo::new("todo 3".to_string(), id);
         let original_todo_list = create!(Todos, number_todos: 3);
-        let updated_todo_list = create!(Todos, number_todos: 4);
+        let mut updated_todo_list = original_todo_list.clone();
+        updated_todo_list.push(todo_added.clone());
 
-        mock_storage
-            .expect_get_todos_from_filestorage()
-            .return_once(|| Ok(original_todo_list));
+        mock_storage.expect_add_todo().return_once(|_| Ok(()));
```

Comment:
Nos testes do mockall, é importante usar o `.times(1)` pra garantir que a função seja executada exatamente uma vez. O `return_once` é misleading, ele permite executar _até_ uma vez, mas não define número mínimo de chamadas.
```suggestion
        mock_storage.expect_add_todo().times(1).returning(|_| Ok(()));
```

Reply:
Mesma coisa nos outros `expect`s. Você vai notar que 75% dos testes da nossa codebase não tem o `.times`/`.once`